src/WtsApi32.Desktop/WtsApi32+WtsSafeMemoryGuardIterator.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <content>
+    /// Exported functions from the WtsApi32.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class WtsApi32
+    {
+        /// <summary>
+        /// Iterator for iterating through memory returned by WtsApi32
+        /// </summary>
+        /// <typeparam name="TIteratorType">Element type</typeparam>
+        internal struct WtsSafeMemoryGuardIterator<TIteratorType> : IEnumerator<TIteratorType>
```

Comment:
These higher API-level types are usually more appropriate for another library. I recommend you remove it.